docs/spec/spec.md

```diff
@@ -3990,6 +4033,75 @@ service /graphql on new graphql:Listener(9090) {
 
 In this example, caching is enabled at the operation level. Therefore, the field `name` and `type` will be cached. Since the field-level cache configuration overrides the parent cache configurations, the field `version` will not be cached. When updating the name with a mutation, the cached values become invalid. Hence, the `invalidate` function can be used to invalidate the existing cache values.
 
+#### 10.7.2 Document Caching
+
+The Ballerina GraphQL module supports server-side document caching, which can be enabled through service configuration. The Ballerina cache module manages the cache internally using a single in-memory cache table. The GraphQL engine automatically generates cache keys by hashing the document. Before hashing, the engine normalizes the document by removing unnecessary elements such as whitespace, new lines, and comments while preserving its logical structure. This ensures consistent cache keys for equivalent queries, regardless of formatting differences.
+
+The internal cache table uses an LRU eviction policy, and cache entries remain valid until they are evicted by the LRU policy. Once the cache hits, it skips the document parsing process.
+
+> **Note**: Documents are cached only if they are parsed successfully without errors.
+
+#### 10.7.2.1 Enable Document Caching
+
+The document caching can be enabled by providing the configuration via the GraphQL service annotation. The field `documentCachConfig` accepts the configurations as described in the [document cache configuration](#7110-document-cache-configurations) section.
+
+###### Example: Document Caching
+
+```ballerina
+import ballerina/graphql;
+
+public type Profile record {|
+    string name;
+    int age;
+|};
+
+@graphql:ServiceConfig {
+    documentCacheConfig: {
+        maxSize: 20
+    }
+}
+service /graphql on new graphql:Listener(9090) {
+    resource function get profile() returns Profile {
+        return {
+            name: "Walter White",
+            age: 51
+        };
+    }
+}
+```
+
+#### 10.7.2.2 Recommended Best Practices
```

Comment:
Shall we include a section under this for cache size? 
We should explain the cache size is measured in number of entries, not by size, so they have to consider that?